app/models/ability.rb

```diff
@@ -29,4 +31,12 @@ def initialize(user)
       can :read, ActiveAdmin
     end
   end
+
+  private
+
+  MINIMUM_DAYS_TO_CANCEL = 7
+
+  def within_cancel_range?(vacation)
```

Comment:
Maybe `approved_within_cancel_range?` is more clear.

Reply:
or add Vacation#cancelable? and call here in the ability

Reply:
I've made the changes, but I'm not sure if I did it right or not.  If you guys find any improvements, just let me know.